libraries/AP_Motors/AP_MotorsCoax.cpp

```diff
@@ -39,6 +39,8 @@ void AP_MotorsCoax::init(motor_frame_class frame_class, motor_frame_type frame_t
 
     // record successful initialisation if what we setup was the desired frame_class
     set_initialised_ok(frame_class == MOTOR_FRAME_COAX);
+
+    _mav_type = MAV_TYPE_COAXIAL;
```

Comment:
I suspect the set_initialised_ok() should be the last thing that is called although it may not matter functionally

Reply:
I have re-ordered